.github/workflows/zephyr.yml

```diff
@@ -54,3 +54,56 @@ jobs:
         run: cd workspace && ./sof/zephyr/docker-run.sh
              ./sof/zephyr/docker-build.sh --cmake-args=-DEXTRA_CFLAGS=-Werror
              --cmake-args=--warn-uninitialized ${{ matrix.IPC_platforms }}
+
+
+  # The sparse build has the worst error handling ever.
+  # Decoder ring as of Sep 2022:
+
+  # CMake Error at zephyr/cmake/compiler/gcc/target.cmake:17 (message):
+  # C compiler
+  # ZSDK/xtensa-intel_s1000_zephyr-elf/bin/xtensa-intel_s1000_zephyr-elf-gcc
+  # not found - Please check your toolchain installation
+  # Call Stack (most recent call first):
+  #   zephyr/cmake/modules/target_toolchain.cmake:63 (include)
+  #   zephyr/cmake/modules/zephyr_default.cmake:121 (include)
+  #   zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
+  #   zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boil...
+  #   CMakeLists.txt:5 (find_package)
+  #
+  #   =>  means "sparse was not found in your PATH"
+  #       Note the "not found" file actually exists!
+
+  #  modules/hal/xtensa/include/xtensa/config/core.h:54:10:
+  #     error:unable to open 'core-isa.h'`
+  #
+  #    => translates to "REAL_CC is not defined or wrong"
```

Comment:
> The sparse build has the worst error handling ever.

So we need to improve it, not build even worst solutions on top of something bad.
We are already very bad on projects integrations, lets not degrade further if possible.
Please go, submit a PR to Zephyr project, add `find_program` for sparse executable, then error message displayed will have proper content (that sparse executable had not been found).